girder/api/rest.py

```diff
@@ -298,9 +298,10 @@ def endpointDecorator(self, *args, **kwargs):
             if 'Content-Range' in cherrypy.response.headers:
                 cherrypy.response.status = 206
 
-            if isinstance(val, types.FunctionType):
-                # If the endpoint returned a function, we assume it's a
-                # generator function for a streaming response.
+            if callable(val):
+                # If the endpoint returned anything callable (function,
+                # lambda, functools.partial), we assume it's a generator
+                # function for a streaming response.
```

Comment:
This change makes it so the below block will never be entered (`if isinstance(val, cherrypy.lib.file_generator)`), which would break any instance where someone used the cherrypy explicit static file methods. We can fix that by simply moving the second block above this one.